mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -7,141 +7,413 @@
 #       extension: .py
 #       format_name: percent
 #       format_version: '1.3'
-#       jupytext_version: 1.13.3
+#       jupytext_version: 1.13.4
 #   kernelspec:
 #     display_name: 'Python 3.8.12 64-bit (''mapping_parenting_tech'': conda)'
 #     language: python
 #     name: python3
 # ---
 
 # %% [markdown]
-# # Code to extract app ids from [list of] web pages saved from Google Play store
+# # Extract app ids from web pages saved from Google Play store
+
+# %% [markdown]
+# Code below carries out the following set of functions:
+# 1. retrieves the ids of apps from a Play Store web page that has been saved locally
+# 2. saves and loads app ids to/from a pickle file so they can be retrieved and used later
+# 3. downloads details for an app/s using its/their app id/s
+# 4. saves and loads app details so they can be retrieved and used later
+# 5.
 
 # %% [markdown]
 # Do imports and set file locations
 
 # %%
-from google_play_scraper import Sort, app, reviews_all
-from tqdm import tqdm
 import re
 import json
+import pickle
+import csv
+from google_play_scraper import Sort, app, reviews
+from tqdm import tqdm
+from pathlib import Path
 from mapping_parenting_tech import PROJECT_DIR, logging
 
-INPUT_PATH = PROJECT_DIR / "inputs/data"
-OUTPUT_PATH = PROJECT_DIR / "outputs/data"
-STORE_PATH = "play_store/"
-PLAY_DATA_FILE = "playstore_innerHTML.html"
-OUTPUT_PLAY_IDS_FILE = "play_store_ids.json"
-OUTPUT_PLAY_DETAILS_FILE = "play_store_details.json"
-OUTPUT_PLAY_REVIEWS_FILE = "play_store_reviews.json"
 
-input_data = INPUT_PATH / STORE_PATH / PLAY_DATA_FILE
+# %% [markdown]
+# ## Read HTML and extract list of app ids
+
+# %%
+def get_playstore_app_ids(filename: str) -> list:
+    """
+    Looks inside 'inputs/data' int the project directory for a given HTML file (filename) and scans that file
+    for links containing ids to Play Store apps.
+
+    Argument:
+        filename: str - the file name of the HTML file you wish to get app ids from. Must be in <[project directory]/inputs/data/>
+
+    Returns:
+        List of app ids, each as a str
+    """
+
+    # open the input file
+    with open(PROJECT_DIR / "inputs/data" / filename, "rt") as infile:
```

Comment:
Very minor, but I guess `filename` is more like a path within the `data` folder? (judging by line 96 further below). Perhaps it's more clear if the variable is named something like `filepath`.

Reply:
Point taken and resolved